synapse/handlers/auth.py

```diff
@@ -481,7 +481,7 @@ async def check_ui_auth(
             sid = authdict["session"]
 
         # Convert the URI and method to strings.
-        uri = request.uri.decode("utf-8")
+        uri = request.uri.decode("utf-8")  # type: ignore
```

Comment:
The type of `request.uri` is incorrectly specified in Twisted.

Reply:
Well it isn't "incorrect", but the placeholder is a string and the value is bytes, which is annoying.

Reply:
Maybe make notes on the places where this happens, and submit that issue on `twisted/twisted`?

Reply:
I'm already planning to.